### ISSUE TITLE
Make with-cp display more dev friendly

### ISSUE THREAD

**Issue by arichiardi** [boot-clj/boot#489]
This is a proposal for better make use of the `with-cp` feature. At the moment the display of the conflicts is for instance:
    
    {:conflicts
      {com.taoensso/timbre
        {"4.3.1" (uap-clj), "4.5.1" (com.taoensso/sente)},
         ...

It would be great to have somewhere the whole dep vector displayed for:
a) easy copy and paste in case a user wants to pick the version from the terminal
b) easy processing the required version from a slurp inside other tasks

Maybe something like:

    {:conflicts
      {com.taoensso/timbre
        {[com.taoensso/timbre "4.3.1"] (uap-clj)
         [com.taoensso/timbre "4.5.1"] (com.taoensso/sente)}
         ...

or vector or set of conflicts keyed by the offending parent dependency:

    {:conflicts
       [{'uap-clj [com.taoensso/timbre "4.3.1"] 
         'com.taoensso/sente [com.taoensso/timbre "4.5.1"]}
         ...]}

Better ideas are more then welcome!

**Comment by micha** [comment]
Wouldn't you just do `boot show -p` to see that?

**Comment by arichiardi** [comment]
Oh, does it show conflicts as well? Sorry didn't know that ;)

**Comment by arichiardi** [comment]
Uhm `show -p` does not seem to detect conflicts like `with-cp`:

```
$ boot -P with-cp -w
clojure.lang.ExceptionInfo: Unresolved dependency conflicts.
    data: {:file "/tmp/boot.user214094730605178881.clj", :line 65}
clojure.lang.ExceptionInfo: Unresolved dependency conflicts.
    data: {:conflicts
           {org.jsoup/jsoup
            {"1.7.1"
             (twitter-api twitter-streaming-client/twitter-streaming-client),
             "1.8.3" (io.forward/clojure-mail)}}}
  clojure.core/ex-info/invokeStatic      core.clj: 4617
               clojure.core/ex-info      core.clj: 4617
     boot.task.built-in/fn/fn/fn/fn  built_in.clj:  348
                boot.core/run-tasks      core.clj:  981
                  boot.core/boot/fn      core.clj:  991
clojure.core/binding-conveyor-fn/fn      core.clj: 1938
                                ...                    
$ boot -P show -p
$

```

maybe I am doing something wrong?

**Comment by micha** [comment]
Something is weird there, because that's what `show -p` is supposed to do (the code in `with-cp` for detecting conflicts uses the same functions).

**Comment by arichiardi** [comment]
Found the problem, and now I see a colorful:

```
[✔] reagent
    ✔ 0.6.0-rc
      reagent
    ✘ 0.5.1
      re-frame
[!] cljsjs/react
    ✔ 15.1.0-0
      reagent
    ✘ 0.13.3-1
      re-frame
[✔] org.clojure/tools.reader
    ✘ 1.0.0-beta1
      org.clojure/clojurescript
    ✔ 1.0.0-alpha1
      org.clojure/tools.reader
    ✘ 0.10.0[com.taoensso/timbre "4.3.1"]
      com.taoensso/sente
      org.clojure/tools.namespace
```

However, my goal is to have also something useful and parseable and easy to copy and paste while adapting your `build.boot`, for instance if `with-cp` would give back `[com.taoensso/timbre "4.3.1"]` directly I could just select it in the terminal and paste it. Does it make sense in your opinion?

**Comment by arichiardi** [comment]
Also, the result is not completely the same as `show -p` shows *all* the conflicts while `with-cp` (at least here) is not showing the solved ones.

**Comment by micha** [comment]
The `show -p` issue seems like it's included in #376

**Comment by arichiardi** [comment]
oh nice that's exactly what I am after, closing this then!
